apps/src/templates/JsDebugger.jsx

```diff
@@ -0,0 +1,158 @@
+var connect = require('react-redux').connect;
+
+var i18n = require('../locale');
+var commonStyles = require('../commonStyles');
+var ProtectedStatefulDiv = require('./ProtectedStatefulDiv');
+
+var styles = {
+  noPadding: {
+    padding: 0
+  }
+};
+
+var DebugConsole = function (props) {
+  var classes = 'debug-console';
+  if (!props.debugButtons) {
+    classes += ' no-commands';
+  }
+  if (!props.debugWatch) {
+    classes += ' no-watch';
+  }
+
+  return (
+    <div id="debug-console" className={classes}>
+      <div id="debug-output" className="debug-output"/>
+      <span className="debug-input-prompt">
+        &gt;
+      </span>
+      <div contentEditable spellCheck="false" id="debug-input" className="debug-input"/>
+    </div>
+  );
+};
+
+var DebugButtons = function () {
+  return (
+    <div id="debug-commands" className="debug-commands">
+      <div id="debug-buttons">
+        <span> </span>
+        <button id="pauseButton" className="debugger_button">
+          <img src="/blockly/media/1x1.gif" className="pause-btn icon21"/>
+          {i18n.pause()}
+        </button>
+        <span> </span>
+        <button id="continueButton" className="debugger_button">
+          <img src="/blockly/media/1x1.gif" className="continue-btn icon21"/>
+          {i18n.continue()}
+        </button>
+        <span> </span>
+        <button id="stepOverButton" className="debugger_button">
+          <img src="/blockly/media/1x1.gif" className="step-over-btn icon21"/>
+          {i18n.stepOver()}
+        </button>
+
+        <button id="stepOutButton" className="debugger_button">
+          <img src="/blockly/media/1x1.gif" className="step-out-btn icon21"/>
+          {i18n.stepOut()}
+        </button>
+        <span> </span>
+        <button id="stepInButton" className="debugger_button">
+          <img src="/blockly/media/1x1.gif" className="step-in-btn icon21"/>
+          {i18n.stepIn()}
+        </button>
+      </div>
+    </div>
+  );
+};
+
+var DebugWatch = function (props) {
+  var classes = 'debug-watch';
+  if (!props.debugButtons) {
+    classes += 'no-commands';
+  }
+  return (
+    <div id="debug-watch" className={classes}/>
+  );
+};
```

Comment:
:+1: Would be nice to also break slider-cell into one of these, even if we don't put it in a new file _yet._